packages/editor/src/plugins/solution/editor.tsx

```diff
@@ -37,58 +67,77 @@ export function SolutionEditor({ state, focused }: SolutionProps) {
-              onChange={(event) => {
-                const newValue = event.target.value.replace(/[^0-9]/g, '')
-                if (prerequisite.defined) {
-                  prerequisite.id.set(newValue)
-                } else {
+            {prerequisite.title.value}
+          </a>
+        ) : (
+          <SerloAddButton
+            text={solutionStrings.addPrerequisite}
+            onClick={() => setShowPrerequisiteLinkTool(true)}
+            className="mb-8 mt-4"
+          />
+        )}
+      </>
+    )
+  }
+
+  function renderPrequisiteTool() {
+    return (
+      <div
+        className="absolute bottom-0 left-0 right-0 top-0"
+        onClick={() => setShowPrerequisiteLinkTool(false)}
+      >
+        <div
+          className="absolute left-12 top-28 z-[95] whitespace-nowrap"
+          onClick={(e) => e.stopPropagation()}
+        >
+          <div
+            className="w-[460px] rounded bg-white text-start not-italic shadow-menu"
+            style={{ width: `${linkOverlayWrapperWidth}px` }}
+          >
+            {prerequisite.defined ? (
+              <LinkOverlayWithHref
+                value={prerequisite.defined ? `/${prerequisite.id.value}` : ''}
+                removeLink={() => {
+                  prerequisite.remove()
+                  setShowPrerequisiteLinkTool(false)
+                }}
+                quickbarData={quickbarData}
+              />
+            ) : (
+              <LinkOverlayEditMode
+                isSerloLinkSearchActive //
+                setHref={(href, title) => {
+                  if (!title) {
+                    showToastNotice('Please choose an existing serlo content')
+                    return
+                  }
                   prerequisite.create({
-                    id: newValue,
-                    title: '',
+                    id: href.replace('/', ''),
+                    title,
                     alias: undefined,
                   })
-                }
-              }}
-            />
-            <a
-              target="_blank"
-              href={hasId ? `/${prerequisite.id.value}` : ''}
-              rel="noopener noreferrer"
-            >
-              <span title={solutionStrings.openArticleTab} className="ml-2.5">
-                <FaIcon icon={faUpRightFromSquare} />
-              </span>
-            </a>
-          </InlineSettings>
-        ) : null}
-        <a className="serlo-link">
-          <InlineInput
-            value={prerequisite.defined ? prerequisite.title.value : ''}
-            onChange={(value) => {
-              if (prerequisite.defined) {
-                prerequisite.title.set(value)
-              } else {
-                prerequisite.create({ id: '', title: value, alias: undefined })
-              }
-            }}
-            placeholder={solutionStrings.linkTitle}
-          />
-        </a>
-      </>
+                }}
+                removeLink={() => {}} //
```

Comment:
Should this really be just `() => {}`?

Reply:
Jup, because there is no link defined at this point so I can't delete it.
I'll make `removeLink` optional to get rid of this weird prop.